### PR TITLE
View todays slots only

### PR DESCRIPTION
**Pull request by szabolcs-keresztes**
## Description and relevant Jira numbers

As we get all journal slots at once we need to group them by date so that we are able to select the different list of slots for each day. 
The grouping has been implemented and used in effects
Also the `SlotProvider` has been implemented which detects the changes and groups the slots by date.

## Pull Request checklist:

- [ ] [WIP] tag removed from PR title
- [ ] PR has an understandable description

## Git feature branch checklist:

- [x] branch name comply with our branching strategy
- [ ] git branch contains relevant JIRA ticket number
- [ ] branch rebased against the latest develop
- [ ] commits are squashed

## Sign off process checklist:

- [ ] Code has been tested manually
- [ ] Tests are passing
- [ ] PR link added to JIRA
- [ ] Reviewers selected in Github
- [ ] Any new reusable component/widget added to component library on Confluence

- [ ] Screenshot(s) captured on the physical device (if applicable)
- [ ] Tested by QA
- [ ] PO's approval

### CODE REVIEW

File: src/providers/slot/slot.ts [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+        if (Array.isArray(differenceToSlot) && differenceToSlot.some(change => change.kind === 'E')) {
+          differenceFound = true;
+        }
+      }
+  
+      return new SlotItem(newSlot, differenceFound)
+    });
+  }
+
+  groupByDate(slots: any[]): {[k: string]: SlotItem[]} {
```
Comment (ammarhaiderbjss) [inline src/providers/slot/slot.ts]: thinking about it, why not just use https://lodash.com/docs/4.17.11#groupBy?

**Comment by ammarhaiderbjss** [comment]
Can we please migrate the existing test from `newSlotsDetectingChanges.spec` into the new slot provider? Also need a test for the new `groupByDate` method

File: src/providers/slot/slot.ts [hunk 2]
```diff
@@ -0,0 +1,54 @@
+import { Injectable } from '@angular/core';
+import { DeepDiff } from 'deep-diff';
+import { flatten } from 'lodash';
+import * as moment from 'moment';
+
+import { SlotItem } from '../slot-selector/slot-item';
+import { ExaminerWorkSchedule } from '../../common/domain/DJournal';
+
+@Injectable()
+export class SlotProvider {
```
Comment (ammarhaiderbjss) [inline src/providers/slot/slot.ts]: `SlotProvider` as a name is perhaps a bit generic, could it be something a bit more descriptive like `SlotUtilitiesProvider`?

File: src/providers/slot/slot.ts (hunk 1 above)
Comment (szabolcs-keresztes) [reply src/providers/slot/slot.ts]: Good point, it was fun to build it I guess

Comment (szabolcs-keresztes) [reply src/providers/slot/slot.ts]: changed it

File: src/pages/journal/journal.effects.ts [hunk 3]
```diff
@@ -32,8 +36,13 @@ export class JournalEffects {
       return this.journalProvider
         .getJournal(journal.lastRefreshed)
         .pipe(
-          map(journalData => newSlotsDetectingChanges(journal.slots, journalData)),
-          map(slots => new journalActions.LoadJournalSuccess(slots)),
+          map((journalData: ExaminerWorkSchedule) => this.slotProvider.detectSlotChanges(journal.slots, journalData)),
+          map((slots: any[]) => {
+            const s = groupBy(slots, this.slotProvider.getSlotDate);
+            console.log('slots', s);
```
Comment (johnmorleybjss) [inline src/pages/journal/journal.effects.ts]: Log line to be removed + map block can be onelined into return value.

Comment (szabolcs-keresztes) [reply src/pages/journal/journal.effects.ts]: ahh thanks for pointing it out. Thought I got rid of all of them :D

**Pull request merged**